src/math/integerhistogram1d.cpp

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "math/integerhistogram1d.h"
+#include "base/lineparser.h"
+#include "base/messenger.h"
+#include "base/sysfunc.h"
+#include "templates/algorithms.h"
+#include <cassert>
+
+IntegerHistogram1D::IntegerHistogram1D()
+{
+    accumulatedData_.addErrors();
+
+    clear();
+}
+
+IntegerHistogram1D::IntegerHistogram1D(const IntegerHistogram1D &source) { (*this) = source; }
+
+// Clear Data
+void IntegerHistogram1D::clear()
+{
+    minimum_ = 0;
+    maximum_ = 0;
```

Comment:
Need to be careful here, since this will reset the limits to an unusable range. 
```suggestion
    minimum_ = 0;
    maximum_ = std::nullopt;
```